### PR TITLE
Add bag inferences for operators: intersection, duplicate_removal, and empty bags

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR adds inferences for operators: intersection, duplicate_removal, and empty bags during post check. 
It also fixes a bug in SolverState::getElements

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
Can you revert your changes to theory_engine.cpp?  It seems to have been unintentionally formatted.

**Comment by mudathirmahgoub** [comment]
> Can you revert your changes to theory_engine.cpp? It seems to have been unintentionally formatted.

Yes. I didn't realize it until I made the pull request. It is tedious to revert it back with other changes

File: src/theory/bags/solver_state.h [hunk 1]
```diff
@@ -77,6 +85,8 @@ class SolverState : public TheoryState
   std::set<Node> d_bags;
   /** bag -> associated elements */
   std::map<Node, std::set<Node>> d_bagElements;
+  /** Disequal bag terms */
+  std::set<Node> d_deq;
```
Comment (ajreynol) [inline src/theory/bags/solver_state.h:89]: `std::vector` is better for this (its faster).  Same applies for d_bags.

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor request, otherwise looks good to me.

**Review by ajreynol — APPROVED** [review]
Discussed offline, LGTM

**Pull request merged**